src/dvbcsa_bs_avx2.h

```diff
@@ -166,5 +150,59 @@ static inline void block_sbox_avx3(dvbcsa_bs_word_t *src, dvbcsa_bs_word_t *dst)
 
 #define BLOCK_SBOX(in_buf, out_buf) block_sbox_avx2(in_buf, out_buf);
 
+/* block cipher 2-word load with byte-deinterleaving */
+// BS_LOAD_DEINTERLEAVE_8 replacement that only loads data that are already uninterleaved
+
+#define BS_LOAD_DEINTERLEAVE_8(ptr, var_lo, var_hi) \
+      {\
+      var_lo = _mm256_load_si256((ptr)); \
+      var_hi = _mm256_load_si256((ptr) + 1); \
+      }
+
+extern const uint16_t dvbcsa_block_sbox_perm[256];
+static inline void block_sbox_permute_interleave_avx(dvbcsa_bs_word_t *src, dvbcsa_bs_word_t *dst) {
```

Comment:
I would name it `avx2_block_sbox_permute` or `block_sbox_permute_avx2`. It doesn't interleave anything.